src/Microsoft.AspNetCore.Server.Testing/Deployers/RemoteDeployer/RemoteDeployer.cs

```diff
@@ -0,0 +1,276 @@
+                && _deploymentParameters.ServerType != ServerType.Kestrel
+                && _deploymentParameters.ServerType != ServerType.WebListener)
+            {
+                throw new InvalidOperationException($"Server type {_deploymentParameters.ServerType} is not supported for remote deployment." +
+                    $" Supported server types are {nameof(ServerType.Kestrel)}, {nameof(ServerType.IIS)} and {nameof(ServerType.WebListener)}");
+            }
+
+            if (string.IsNullOrWhiteSpace(_deploymentParameters.ServerName))
+            {
+                throw new ArgumentException($"Invalid value '{_deploymentParameters.ServerName}' for {nameof(RemoteDeploymentParameters.ServerName)}");
+            }
+
+            if (string.IsNullOrWhiteSpace(_deploymentParameters.ServerAccountName))
+            {
+                throw new ArgumentException($"Invalid value '{_deploymentParameters.ServerAccountName}' for {nameof(RemoteDeploymentParameters.ServerAccountName)}." +
+                    " Account credentials are required to enable creating a powershell session to the remote server.");
+            }
+
+            if (string.IsNullOrWhiteSpace(_deploymentParameters.ServerAccountPassword))
+            {
+                throw new ArgumentException($"Invalid value '{_deploymentParameters.ServerAccountPassword}' for {nameof(RemoteDeploymentParameters.ServerAccountPassword)}." +
+                    " Account credentials are required to enable creating a powershell session to the remote server.");
+            }
+
+            if (string.IsNullOrWhiteSpace(_deploymentParameters.RemoteServerFileShare))
+            {
+                throw new ArgumentException($"Invalid value for {nameof(RemoteDeploymentParameters.RemoteServerFileShare)}." +
+                    " . A file share is required to copy the application's published output.");
+            }
+
+            if (string.IsNullOrWhiteSpace(_deploymentParameters.RemoteServerRelativeExecutablePath))
+            {
+                throw new ArgumentException($"Invalid value for {nameof(RemoteDeploymentParameters.RemoteServerRelativeExecutablePath)}." +
+                    " This is the name of the executable in the published output which needs to be executed on the remote server.");
+            }
+
+            if (string.IsNullOrWhiteSpace(_deploymentParameters.ApplicationBaseUriHint))
+            {
+                throw new ArgumentException($"Invalid value for {nameof(RemoteDeploymentParameters.ApplicationBaseUriHint)}.");
+            }
+        }
+
+        public override DeploymentResult Deploy()
+        {
+            if (_isDisposed)
+            {
+                throw new ObjectDisposedException("This instance of deployer has already been disposed.");
+            }
+
+            // Publish the app to a local temp folder on the machine where the test is running
+            DotnetPublish();
+
+            var folderId = Guid.NewGuid().ToString();
```

Comment:
You still have the problem that if you call Deploy twice only the second one will be disposed on Dispose